### PR TITLE
Moving the Fixes from internal AvroRelConverter

### PR DESCRIPTION
**Pull request by srinipunuru**
Misc fixes that are made in internal copy of AvroRelConverter.

### CODE REVIEW

File: samza-sql/src/main/java/org/apache/samza/sql/avro/AvroRelConverter.java [hunk 1]
```diff
@@ -170,22 +179,22 @@ static public GenericRecord convertToGenericRecord(SamzaSqlRelRecord relRecord,
               schema.getName(), schema.getNamespace(), fieldName);
           continue;
         }
-
         Object relObj = values.get(index);
+        if (relObj == null) {
```
Comment (shenodaguirguis) [inline samza-sql/src/main/java/org/apache/samza/sql/avro/AvroRelConverter.java]: ?

File: samza-sql/src/main/java/org/apache/samza/sql/avro/AvroRelConverter.java [hunk 2]
```diff
@@ -271,45 +281,46 @@ public Object convertToJavaObject(Object avroObj, Schema schema) {
     }
   }
 
-  private boolean isSchemaCompatible(Object avroObj, Schema unionSchema) {
-    if (unionSchema.getType() == Schema.Type.NULL && avroObj == null) {
-      return true;
+  private static boolean isSchemaCompatibleWithRelObj(Object relObj, Schema unionSchema) {
+    if (unionSchema.getType() == Schema.Type.NULL) {
```
Comment (shenodaguirguis) [inline samza-sql/src/main/java/org/apache/samza/sql/avro/AvroRelConverter.java]: verify/handle unionSchema != null?

File: samza-sql/src/main/java/org/apache/samza/sql/avro/AvroRelConverter.java [hunk 3]
```diff
@@ -271,45 +281,46 @@ public Object convertToJavaObject(Object avroObj, Schema schema) {
...
+        return relObj instanceof Float || relObj instanceof Double;
       default:
         return true;
     }
   }
 
-  private static boolean isSchemaCompatibleWithRelObj(Object relObj, Schema unionSchema) {
-    if (unionSchema.getType() == Schema.Type.NULL && relObj == null) {
+  private static boolean isSchemaCompatible(Object avroObj, Schema unionSchema) {
+    if (unionSchema.getType() == Schema.Type.NULL && avroObj == null) {
```
Comment (shenodaguirguis) [inline samza-sql/src/main/java/org/apache/samza/sql/avro/AvroRelConverter.java]: shouldn't (avroObj == null) move to be the return?

File: samza-sql/src/main/java/org/apache/samza/sql/avro/AvroRelConverter.java (hunk 1 above)
Comment (srinipunuru) [reply samza-sql/src/main/java/org/apache/samza/sql/avro/AvroRelConverter.java]: Dead code, needs to be removed.

File: samza-sql/src/main/java/org/apache/samza/sql/avro/AvroRelConverter.java (hunk 2 above)
Comment (srinipunuru) [reply samza-sql/src/main/java/org/apache/samza/sql/avro/AvroRelConverter.java]: I believe you are asking whether we should handle null values for unionSchema?

Comment (shenodaguirguis) [reply samza-sql/src/main/java/org/apache/samza/sql/avro/AvroRelConverter.java]: I was actually wondering if we should guard against NullPointerException when  unionSchema of Schema is null

**Pull request merged**